[PATCH] add kmap API abstraction

This patch allows the kmap/kunmap API to be abstracted for implementation
purposes by architectures that don't implement highmem.  The purpose
of this is for those architectures that need to perform arch specific
coherence operations on kmap/kunmap.

Signed-off-by: James Bottomley <James.Bottomley@SteelEye.com>
Index: linux-2.6/include/linux/highmem.h
===================================================================
--- linux-2.6.orig/include/linux/highmem.h	2006-07-26 17:51:09.000000000 -0700
+++ linux-2.6/include/linux/highmem.h	2006-07-26 17:51:18.000000000 -0700
@@ -29,6 +29,7 @@
 
 static inline unsigned int nr_free_highpages(void) { return 0; }
 
+#ifndef ARCH_HAS_KMAP
 static inline void *kmap(struct page *page)
 {
 	might_sleep();
@@ -41,6 +42,7 @@
 #define kunmap_atomic(addr, idx)	do { } while (0)
 #define kmap_atomic_pfn(pfn, idx)	page_address(pfn_to_page(pfn))
 #define kmap_atomic_to_page(ptr)	virt_to_page(ptr)
+#endif
 
 #endif /* CONFIG_HIGHMEM */
 
